### ISSUE TITLE
Promiselike API

### ISSUE THREAD

**Issue by JohnRodney** [EddyVerbruggen/Calendar-PhoneGap-Plugin#275]
@EddyVerbruggen Hey I was wondering if you would be interested in a promise wrapper I've written for your JS.  I built out an es6 class that wraps all the callbacks in promises to get around the problem of trying to call multiple createEvents at the same time.

Since I used all JS to do so then it shouldn't cause any breaking changes to the underlying hood at all.  The benefits is now I can do a reduce on a large set of events and send them into the calendar without worrying about the callback style syntax. 

EX Usage: 

```
eventArray.reduce((curr, next) => {
  return curr.then(() => {
    return window.plugins.calendar.asyncCreateEventWithOptions(next, options);
  });
}, P.resolve())
  .catch((err) => { console.log('creating events failed with:', err); });
```

you will notice I just pass next, and options as the async event uses object destructuring so you can pass it any event object that looks like:
```
{ 
  title: 'someTitle', 
  notes: 'some notes', 
  location: 'some location', 
  startDate: aDateObject, 
  endDate: aDateObject 
}
``` 
and a valid options object.  In my wrapper I did this so far to only the functionality supported by both android and IOS, but it could be updated to support all internal parts of the api.

Strengths IMO: This takes away the need of the user to put together some sort of structure when dealing with a large set of events they need to do actions on the native calendar with.

If you are interested in this type of feature being added to the repo please get back to me and I will discuss the more minute details with you and work on it.

This is in reference to: Issues such as #234, and any other issues that may arise while people are trying to manage a high number of callbacks.

**Comment by EddyVerbruggen** [comment]
Sounds very cool! Perhaps it's sufficient to add a few code snippets to the readme, or a helper class which users may or may not link to in their projects? As long as we're not forcing anyone to use promises I'm on board.

**Comment by JohnRodney** [comment]
@EddyVerbruggen alright since it took a while to hear from ya I am not working on this at the moment.  I will be revisiting my calendar plugin for that app I had developed this for in a few weeks and I may see if I can refactor it into a PR.  Thanks.

**Comment by JohnRodney** [comment]
@EddyVerbruggen I ended up making a seperate repository that can be applied as a wrapper to this repo for those who need / want promises along with error checking before passing the payloads to the plugin.  I ran into alot of situations where the internals weren't validating objects first to figured it might be useful.  https://github.com/poetic-labs/native-calender-api

You are welcome to add it to the end of the readme.
